sympy/utilities/tests/test_codegen.py

```diff
@@ -1441,3 +1441,29 @@ def test_global_vars():
                          global_vars=(z, t))
         source = result[0][1]
         assert source == expected
+
+def test_custom_codegen():
+    from sympy.printing.ccode import C99CodePrinter
+    from sympy.functions.elementary.exponential import exp
+
+    printer = C99CodePrinter(settings={'user_functions': {'exp': 'fastexp'}})
+    gen = C99CodeGen(printer=printer)
+    gen.preprocessor_statements.append('#include "fastexp.h"\n')
```

Comment:
Ok they're appended automatically now. It doesn't really hurt anything if the user wants to have a more complicated preprocessor statement.